sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -173,25 +188,22 @@ private[csv] class UnivocityParser(
    */
   def parse(input: String): Option[InternalRow] = {
     convertWithParseMode(parser.parseLine(input)) { tokens =>
-      var i: Int = 0
-      while (i < indexArr.length) {
-        val pos = indexArr(i)
+      indexArr.foreach { case (pos, i) =>
```

Comment:
(Oh, BTW, we should just use `while` in the critical path if we are not pretty sure whether the byte codes are virtually the same or more efficient. This is also written as `Use while loops instead of for loops or functional transformations (e.g. map, foreach)` in [scala-style-guide#traversal-and-zipwithindex](https://github.com/databricks/scala-style-guide#traversal-and-zipwithindex))